File: frontend/src/objects/order/PartialRefundStatus.ts
// Business note: frontend mirror of a backend service object; keep names and fields aligned unless this file is explicitly frontend-only.
export const PARTIAL_REFUND_STATUS = {
  pending: 'Pending',
  approved: 'Approved',
  rejected: 'Rejected',
} as const

export type PartialRefundStatus =
  (typeof PARTIAL_REFUND_STATUS)[keyof typeof PARTIAL_REFUND_STATUS]

File: backend/src/main/scala/services/order/objects/PartialRefundStatus.scala
package services.order.objects

// Business note: service-owned domain object; mirror it in the frontend only when it is part of the protocol surface.
import system.objects.given
import system.objects.*

import io.circe.{Decoder, Encoder}

enum PartialRefundStatus:
  case Pending, Approved, Rejected

object PartialRefundStatus:
  private val enumLabel = text("PartialRefundStatus")
  def render(value: PartialRefundStatus): DisplayText = enumDisplayText(value)
  def parse(value: DisplayText): Option[PartialRefundStatus] = parseEnumValue(value, PartialRefundStatus.values)
  given Encoder[PartialRefundStatus] = enumEncoder
  given Decoder[PartialRefundStatus] = enumDecoder(PartialRefundStatus.values, enumLabel)
